actool: crash on patch-manifest

$ actool patch-manifest --exec=/bin/bash alpine.aci alpine2.aci

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x407b3b]

goroutine 1 [running]:
main.patchManifest(0xc208070000, 0x0, 0x0)
	/build/amd64-usr/var/tmp/portage/app-emulation/rkt-0.7.0-r1/work/rkt-0.7.0/build-rkt-0.7.0/gopath/src/github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/actool/manifest.go:149 +0x196b
main.extractManifest(0xc20803efc0, 0xc20806ed80, 0xc20803ef00, 0x0, 0x0, 0x0, 0x0, 0x0)
	/build/amd64-usr/var/tmp/portage/app-emulation/rkt-0.7.0-r1/work/rkt-0.7.0/build-rkt-0.7.0/gopath/src/github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/actool/manifest.go:275 +0x61d
main.runPatchManifest(0xc20800a030, 0x2, 0x2, 0x0)
	/build/amd64-usr/var/tmp/portage/app-emulation/rkt-0.7.0-r1/work/rkt-0.7.0/build-rkt-0.7.0/gopath/src/github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/actool/manifest.go:425 +0xd81
main.main()
	/build/amd64-usr/var/tmp/portage/app-emulation/rkt-0.7.0-r1/work/rkt-0.7.0/build-rkt-0.7.0/gopath/src/github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/actool/actool.go:100 +0x60f
co
```

--- a/actool/manifest.go
+++ b/actool/manifest.go
@@ -145,23 +145,31 @@ func patchManifest(im *schema.ImageManifest) error {
 		im.Name = *name
 	}
 
+	var app *types.App = im.App
 	if patchExec != "" {
-		im.App.Exec = strings.Split(patchExec, " ")
+		if app == nil {
+			// if the original manifest was missing an app and
+			// patchExec is set let's assume the user is trying to
+			// inject one...
+			app = &types.App{}
+		}
+		app.Exec = strings.Split(patchExec, " ")
+	}
+
+	if patchUser != "" || patchGroup != "" || patchCaps != "" || patchMounts != "" || patchPorts != "" || patchIsolators != "" {
+		// ...but if we still don't have an app and the user is trying
+		// to patch one of its other parameters, it's an error
+		if app == nil {
+			return fmt.Errorf("no app in the supplied manifest and no exec command provided")
+		}
 	}
 
 	if patchUser != "" {
-		im.App.User = patchUser
-	}
-	if patchGroup != "" {
-		im.App.Group = patchGroup
+		app.User = patchUser
 	}
 
-	var app *types.App
-	if patchCaps != "" || patchMounts != "" || patchPorts != "" || patchIsolators != "" {
-		app = im.App
-		if app == nil {
-			return fmt.Errorf("no app in the manifest")
-		}
+	if patchGroup != "" {
+		app.Group = patchGroup
 	}
 
 	if patchCaps != "" {

